src/TagField.php

```diff
@@ -298,6 +312,21 @@ public function setValue($value, $source = null)
         return parent::setValue(array_filter($value));
     }
 
+    /**
+     * Gets the source array if required
+     *
+     * Note: this is expensive for a SS_List
+     *
+     * @return array
+     */
+    public function getSource()
+    {
+        if (is_null($this->source)) {
```

Comment:
Above you overwrite the default `$source` to just an empty array but you're checking for null here. This means consumers of `getSource()` will now just get empty arrays?

Reply:
My fix addresses this in setSource (it'll set to null if we have a DataList).